packages/wrangler/src/__tests__/metrics.test.ts

```diff
@@ -119,109 +144,401 @@ describe("metrics", () => {
+
+			describe("banner", () => {
+				beforeEach(() => {
+					vi.mocked(getWranglerVersion).mockReturnValue("1.2.3");
+				});
+				it("should print the banner if current version is different to the stored version", async () => {
+					writeMetricsConfig({
+						permission: {
+							enabled: true,
+							date: new Date(2022, 6, 4),
+							bannerLastShown: "1.2.1",
+						},
+					});
+
+					const requests = mockMetricRequest();
+
+					await runWrangler("command subcommand positional");
+					expect(std.out).toMatchInlineSnapshot(`
+						"
+						Cloudflare collects anonymous telemetry about your usage of Wrangler. Learn more at https://github.com/cloudflare/workers-sdk/tree/main/packages/wrangler/telemetry.md
+						Ran wrangler command subcommand"
+					`);
+
+					expect(requests.count).toBe(2);
+				});
+				it("should not print the banner if current version is the same as the stored version", async () => {
+					writeMetricsConfig({
+						permission: {
+							enabled: true,
+							date: new Date(2022, 6, 4),
+							bannerLastShown: "1.2.3",
+						},
+					});
+					const requests = mockMetricRequest();
+					await runWrangler("command subcommand positional");
+					expect(std.out).toMatchInlineSnapshot(`
+						"Ran wrangler command subcommand"
+					`);
+					expect(requests.count).toBe(2);
+				});
+				it("should print the banner if nothing is stored under bannerLastShown and then store the current version", async () => {
+					writeMetricsConfig({
+						permission: {
+							enabled: true,
+							date: new Date(2022, 6, 4),
+						},
+					});
+					const requests = mockMetricRequest();
+					await runWrangler("command subcommand positional");
+					expect(std.out).toMatchInlineSnapshot(`
+						"
+						Cloudflare collects anonymous telemetry about your usage of Wrangler. Learn more at https://github.com/cloudflare/workers-sdk/tree/main/packages/wrangler/telemetry.md
+						Ran wrangler command subcommand"
+					`);
+					expect(requests.count).toBe(2);
+					const { permission } = readMetricsConfig();
+					expect(permission?.bannerLastShown).toEqual("1.2.3");
+				});
+				it("should not print the banner if telemetry permission is disabled", async () => {
+					writeMetricsConfig({
+						permission: {
+							enabled: false,
+							date: new Date(2022, 6, 4),
+						},
+					});
+					const requests = mockMetricRequest();
+					await runWrangler("command subcommand positional");
+					expect(std.out).toMatchInlineSnapshot(`
+						"Ran wrangler command subcommand"
+					`);
+					expect(requests.count).toBe(0);
+					const { permission } = readMetricsConfig();
+					expect(permission?.bannerLastShown).toBeUndefined();
+				});
+			});
+		});
+
+		describe("redactArgValues()", () => {
+			it("should redact sensitive values", () => {
+				const args = {
+					default: false,
+					array: ["beep", "boop"],
+					secretArray: ["beep", "boop"],
+					number: 42,
+					string: "secret",
+					secretString: "secret",
+				};
```

Comment:
Nit: we can expand this to cover the normalize logic:
```suggestion
				const args = {
					default: false,
					array: ["beep", "boop"],
					secretArray: ["beep", "boop"],
					"secret-array":  ["beep", "boop"],
					number: 42,
					string: "secret",
					secretString: "secret",
					"secret-string": "secret",
				};
```

Reply:
ah the normalise logic wasn't actually working - fixed here [f2b0b40](https://github.com/cloudflare/workers-sdk/pull/7291/commits/f2b0b402976dd3cae23b8971b382a025d84c99f4)